examples/terraform/vmware-cloud-director/output.tf

```diff
@@ -0,0 +1,87 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+output "kubeone_api" {
+  description = "kube-apiserver LB endpoint"
+
+  value = {
+    endpoint                    = vcd_vapp_vm.control_plane.0.network.0.ip
+    apiserver_alternative_names = var.apiserver_alternative_names
+  }
+}
+
+output "kubeone_hosts" {
+  description = "Control plane endpoints to SSH to"
+
+  value = {
+    control_plane = {
+      cluster_name         = var.cluster_name
+      cloud_provider       = "vmware-cloud-director"
+      private_address      = vcd_vapp_vm.control_plane.*.network.0.ip
+      hostnames            = vcd_vapp_vm.control_plane.*.name
+      ssh_agent_socket     = var.ssh_agent_socket
+      ssh_port             = var.ssh_port
+      ssh_private_key_file = var.ssh_private_key_file
+      ssh_user             = var.ssh_username
+    }
+  }
+}
+
+output "kubeone_workers" {
+  description = "Workers definitions, that will be transformed into MachineDeployment object"
+
+  value = {
+    # following outputs will be parsed by kubeone and automatically merged into
+    # corresponding (by name) worker definition
+    "${var.cluster_name}-pool1" = {
+      replicas = var.initial_machinedeployment_replicas
+      providerSpec = {
+        sshPublicKeys   = [file(var.ssh_public_key_file)]
+        operatingSystem = var.worker_os
+        operatingSystemSpec = {
+          distUpgradeOnBoot = false
+        }
+        # uncomment to following to set those kubelet parameters. More into at:
+        # https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/
+        # machineAnnotations = {
+        #  "v1.kubelet-config.machine-controller.kubermatic.io/SystemReserved" = "cpu=200m,memory=200Mi"
+        #  "v1.kubelet-config.machine-controller.kubermatic.io/KubeReserved"   = "cpu=200m,memory=300Mi"
+        #  "v1.kubelet-config.machine-controller.kubermatic.io/EvictionHard"   = ""
+        # }
+        cloudProviderSpec = {
+          # provider specific fields:
+          # see example under `cloudProviderSpec` section at:
+          # https://github.com/kubermatic/machine-controller/blob/master/examples/nutanix-machinedeployment.yaml
+          organization     = var.vcd_org_name
+          vdc              = var.vcd_vdc_name
+          vapp             = vcd_vapp.cluster.name
+          catalog          = var.catalog_name
+          template         = var.template_name
+          network          = vcd_vapp_org_network.network.org_network_name
+          cpus             = var.worker_cpus
+          cpuCores         = var.worker_cpu_cores
+          memoryMB         = var.worker_memory
+          diskSizeGB       = var.worker_disk_size
+          storageProfile   = var.worker_disk_storage_profile
+          ipAllocationMode = "DHCP"
```

Comment:
Should we parametrize this?

Reply:
No, we'd want to enforce DHCP here since the networking is configured for DHCP https://github.com/ahmedwaleedmalik/kubeone/blob/vcd-integration/examples/terraform/vmware-cloud-director/main.tf#L115